### DISCUSSION TITLE
Dynamically change useController name

### DISCUSSION THREAD

**Discussion by dbrxnds** [react-hook-form/react-hook-form#9960] [Q&A] [answered]
Hi, dynamically changing a controller name doesn't actually change the value of an input, even though it seems like it should, I believe?

https://codesandbox.io/s/condescending-poincare-vg9e4y?file=/src/App.tsx

**Comment by Moshyfawn** [comment, marked as answer]
The `Controller` name is "cached" and can't be changed

**Reply by dbrxnds** [reply]
Thanks @Moshyfawn that's what I thought. What would you recommend I do in this case?

**Reply by bluebill1049** [reply]
maybe use the `key` prop?

**Reply by dbrxnds** [reply]
Thanks @bluebill1049, that does the trick! I was using `useController` before.
